src/types/basic.ts

```diff
@@ -182,5 +182,83 @@ export const DeviceName = new t.Type<string, string>(
 		),
 	t.identity,
 );
-
 export type DeviceName = t.TypeOf<typeof DeviceName>;
+
+/**
+ * A string boolean, one of ['true', 'false', '1', '0', 'on', 'off'].
+ * Decodes to a boolean.
+ */
+
+export const BooleanFromString = new t.Type<boolean, string, unknown>(
+	'BooleanFromString',
+	t.boolean.is,
+	(i, c) =>
+		pipe(
+			t.string.validate(i, c),
+			chain((s) => {
+				if (['true', '1', 'on'].includes(s)) {
+					return t.success(true);
+				}
+
+				if (['false', '0', 'off'].includes(s)) {
+					return t.success(false);
+				}
+
+				return t.failure(i, c, 'must be a valid string boolean');
+			}),
+		),
+	String,
+);
+export type BooleanFromString = t.TypeOf<typeof BooleanFromString>;
+
+/**
+ * A boolean integer, one of [0, 1].
+ * Decodes to a boolean.
+ */
+export const BooleanFromInt = new t.Type<boolean, number, unknown>(
+	'BooleanFromInt',
+	t.boolean.is,
+	(i, c) =>
+		pipe(
+			t.number.validate(i, c),
+			chain((n) =>
+				[0, 1].includes(n)
+					? t.success(Boolean(n))
+					: t.failure(i, c, 'must be a valid integer boolean'),
+			),
+		),
+	Number,
+);
+export type BooleanFromInt = t.TypeOf<typeof BooleanFromInt>;
+
+/**
+ * A stringified or regular boolean, one of the set of (VALID_STRING_BOOLS | [true, false] | [0, 1]).
+ * Decodes to a boolean.
+ *
+ * TODO: replace PermissiveBoolean in src/config/types.ts
```

Comment:
I was going to point this out too

Reply:
TODO: Christina will address this by itself in another PR